documentation/jetty-documentation/src/main/java/org/eclipse/jetty/docs/programming/migration/ServletToHandlerDocs.java

```diff
@@ -0,0 +1,643 @@
+
+            // Gets the server name.
+            // Replaces:
+            //   - servletRequest.getServerName()
+            String serverName = Request.getServerName(request);
+
+            // Gets the server port.
+            // Replaces:
+            //   - servletRequest.getServerPort()
+            int serverPort = Request.getServerPort(request);
+
+            // Gets the remote host/address.
+            // Replaces:
+            //   - servletRequest.getRemoteAddr()
+            //   - servletRequest.getRemoteHost()
+            String remoteAddress = Request.getRemoteAddr(request);
+
+            // Gets the remote port.
+            // Replaces:
+            //   - servletRequest.getRemotePort()
+            int remotePort = Request.getRemotePort(request);
+
+            // Gets the local host/address.
+            // Replaces:
+            //   - servletRequest.getLocalAddr()
+            //   - servletRequest.getLocalHost()
+            String localAddress = Request.getLocalAddr(request);
+
+            // Gets the local port.
+            // Replaces:
+            //   - servletRequest.getLocalPort()
+            int localPort = Request.getLocalPort(request);
+
+            // Gets the request attributes.
+            // Replaces:
+            //   - servletRequest.getAttributeNames()
+            //   - servletRequest.getAttribute(name)
+            //   - servletRequest.setAttribute(name, value)
+            //   - servletRequest.removeAttribute(name)
+            String name = "name";
+            Object value = "value";
+            Set<String> names = request.getAttributeNameSet();
+            Object attribute = request.getAttribute(name);
+            request.setAttribute(name, value);
+            request.removeAttribute(name);
+            request.clearAttributes();
+            Map<String, Object> map = request.asAttributeMap();
+
+            // Gets the request trailers.
+            // Replaces:
+            //   - servletRequest.getTrailerFields()
+            HttpFields trailers = request.getTrailers();
+
+            // Gets the HTTP session.
+            // Replaces:
+            //   - servletRequest.getSession()
+            //   - servletRequest.getSession(create)
+            boolean create = true;
+            Session session = request.getSession(create);
+
+            callback.succeeded();
+            return false;
+        }
+    }
+    // end::request[]
+
+    @SuppressWarnings("InnerClassMayBeStatic")
+    public class RequestContentAPIsString extends Handler.Abstract
+    {
+        // tag::requestContent-string[]
+        @Override
+        public boolean handle(Request request, Response response, Callback callback) throws Exception
+        {
+            // Non-blocking read the request content as a String.
+            // Use with caution as the request content may be large.
+            Promise.Completable<String> completable = new Promise.Completable<>();
```

Comment:
Can we do this without Promise?   I find that a very confusing abstraction.  Yeah I know some might be familiar with it, but not generally.
Directly using a `CompletableFuture` would be a more general and understandable example.